quil-py/src/lib.rs

```diff
@@ -44,3 +44,25 @@ macro_rules! impl_to_quil {
         }
     };
 }
+
+#[macro_export]
+macro_rules! impl_eq {
+    ($name: ident) => {
+        #[pyo3::pymethods]
+        impl $name {
+            pub fn __richcmp__(
+                &self,
+                py: pyo3::Python<'_>,
+                other: &Self,
+                op: pyo3::pyclass::CompareOp,
+            ) -> pyo3::PyObject {
+                use pyo3::IntoPy;
+                match op {
+                    pyo3::pyclass::CompareOp::Eq => (self == other).into_py(py),
+                    pyo3::pyclass::CompareOp::Ne => (self != other).into_py(py),
+                    _ => py.NotImplemented(),
```

Comment:
Out of curiosity, is there a specific reason you went for `#[derive(PartialEq)]` and `self == other` instead of the `self.as_inner() == other.as_inner()` that the replaced code had?

Reply:
Mostly because I don't see there being much of a reason to tie the implementation to `PyWrapper` in that way. This makes the macro more generally applicable to anything that can derive `PartialEq`, rather than anything that implements `PyWrapper` on some inner type that derives `PartalEq`.